source/Symbol/SwiftASTContext.cpp

```diff
@@ -3131,21 +3135,39 @@ swift::ASTContext *SwiftASTContext::GetASTContext() {
     if (!clang_importer_options.OverrideResourceDir.empty()) {
       clang_importer_ap = swift::ClangImporter::create(*m_ast_context_ap,
                                                        clang_importer_options);
-      moduleCachePath = swift::getModuleCachePathFromClang(
-          clang_importer_ap->getClangInstance());
-      LOG_PRINTF(LIBLLDB_LOG_TYPES, "Using clang module cache path: %s",
-                 moduleCachePath.c_str());
+      if (clang_importer_ap)
+        moduleCachePath = swift::getModuleCachePathFromClang(
+            clang_importer_ap->getClangInstance());
+      else
+        LOG_PRINTF(LIBLLDB_LOG_TYPES, "Failed to initialize ClangImporter");
     }
   }
 
+  if (moduleCachePath.empty()) {
+    moduleCachePath = GetClangModulesCacheProperty();
+    // Even though it is initialized to the default Clang location at startup a
+    // user could have overwritten it with an empty path.
+    if (moduleCachePath.empty()) {
+      llvm::SmallString<0> path;
+      std::error_code ec =
+          llvm::sys::fs::createUniqueDirectory("ModuleCache", path);
+      if (!ec)
+        moduleCachePath = path.str();
+      else
+        moduleCachePath = "/tmp/lldb-ModuleCache";
```

Comment:
is there a way we can avoid hardcoding the module cache path when we end up fixing this upstream? I think it's OK for the branch, but it would be great if all this knowledge didn't leak in lldb.

Reply:
I don't know what you mean by hardcoding. We are

1. Asking for the user-specified path
2. Asking Clang to compute the default location
3. Aking for a unique temporary directory
4. use /tmp/lldb-ModuleCache

in that order.